code/game/objects/objs.dm

```diff
@@ -64,13 +64,22 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 		breakable_init()
 	if(is_cooktop)
 		add_component(/datum/component/cooktop)
-//Disabled because autoignition would too often turn rooms into firestorms due to a chain reaction
-//where one item would burn up, increase the room temperature (in code/ZAS/Fire.dm, proc/burnFireFuel()) and cause
-//other items to burn up.
-
-//To re-enable, un-comment the lines of code.
-	// if(autoignition_temperature)
-	// 	burnableatoms+=src
+	if(!thermal_mass)
+		switch(w_class)
+			if(W_CLASS_TINY, W_CLASS_SMALL)
+				thermal_mass = 0.1
+			if(W_CLASS_MEDIUM)
+				thermal_mass = 1.0
+			if(W_CLASS_LARGE)
+				thermal_mass = 10.0
+			if(W_CLASS_HUGE)
+				thermal_mass = 25.0 //combo breaker but 100kg is way too heavy
+			if(W_CLASS_GIANT)
+				thermal_mass = 100
+	if(thermal_mass)
+		initial_thermal_mass = thermal_mass //can't just use initial() here as some thermal masses are defined in New()
```

Comment:
Since the supercall is always at the beginning of `New()`, this will be executed before any children update their `thermal_mass` values anyway

Reply:
This will have to be kept in mind if adding any objects which define `thermal_mass` in New(), but for now there are no instances where this is an issue.